zp-relayer/validation/api/validation.ts

```diff
@@ -186,3 +187,44 @@ export const checkGetTransactionsV2 = checkErrors(AjvGetTransactionsV2Schema)
 export const checkGetLimits = checkErrors(AjvGetLimitsSchema)
 export const checkGetSiblings = checkErrors(AjvGetSiblingsSchema)
 export const checkTraceId = checkErrors(AjvTraceIdSchema)
+
+async function fetchSafe(url: string) {
+  const r = await fetch(url)
+  if (!r.ok) {
+    throw new Error(`HTTP status code: ${r.status}`)
+  }
+  return r
+}
+
+export async function validateCountryIP(ip: string) {
+  if (config.blockedCountries.length === 0) return null
+
+  const apis = [
+    fetchSafe(`https://ipapi.co/${ip}/country`).then(res => res.text()),
+    fetchSafe(`https://api.country.is/${ip}`)
+      .then(res => res.json())
+      .then(data => data.country),
+  ]
+  const country: string = await Promise.any(apis).catch(e => {
```

Comment:
Is it possible that two services will return different results for the same IP address?

Reply:
I think, generally, it will be the same in all cases. There is not much info about first API, but the second one uses `MaxMind` data under the hood. Anyway, this approach mimics the one done in UI (for UAE), so, at least, the results will be consistent.